manifests/bridge.pp

```diff
@@ -33,48 +46,79 @@
 # Copyright (C) 2013 Mike Arnold, unless otherwise noted.
 #
 define network::bridge (
-  $ensure,
-  $userctl = false,
-  $stp = false,
-  $delay = '30',
-  $bridging_opts = undef,
-  $ipv6init = false,
-  $restart = true,
+  Enum['up', 'down'] $ensure,
+  Optional[IP::Address::V4::NoSubnet] $ipaddress = undef,
+  Optional[IP::Address::V4::NoSubnet] $netmask = undef,
+  Optional[IP::Address::V4::NoSubnet] $gateway = undef,
+  Optional[IP::Address::V6] $ipv6address = undef,
+  Optional[IP::Address::V6::NoSubnet] $ipv6gateway = undef,
+  Network::If::Bootproto $bootproto = 'none',
+  Boolean $userctl = false,
+  Boolean $stp = false,
+  String $delay = '30',
+  Optional[String] $bridging_opts = undef,
+  Optional[String] $scope = undef,
+  Boolean $peerdns = false,
+  Boolean $ipv6init = false,
+  Boolean $ipv6peerdns = false,
+  Optional[IP::Address::NoSubnet] $dns1 = undef,
+  Optional[IP::Address::NoSubnet] $dns2 = undef,
+  Optional[String] $domain = undef,
+  Boolean $restart = true,
```

Comment:
network::bridge now takes takes all arguments that are used by network::bridge::dynamic and network::bridge::static in order to prevent code duplication.